frontend/packages/topology/src/hooks/useTopologyDataModelFactory.ts

```diff
@@ -0,0 +1,101 @@
+import { useState, useEffect } from 'react';
+import { WatchK8sResources, WatchK8sResourcesGeneric } from '@console/dynamic-plugin-sdk';
+import { CodeRef } from '@console/dynamic-plugin-sdk/src/types';
+
+/**
+ * Resources type that can be:
+ * 1. Static WatchK8sResourcesGeneric (from dynamic plugin SDK)
+ * 2. CodeRef<() => Promise<WatchK8sResourcesGeneric>> (from dynamic plugin SDK, for dynamic resources)
+ * 3. Function (namespace: string) => WatchK8sResources (from old internal plugin system or converted dynamic plugins)
```

Comment:
we should only be supporting 1 and 2, this really should be TopologyDataModelFactory.properties.resources.

you can build on my PR, https://github.com/openshift/console/pull/15617, which removes the final usage of `TopologyDataModelFacotry`. then we no longer care about its support